lib/pry/input_completer.rb

```diff
@@ -239,4 +227,30 @@ def build_path(input)
     end
     return path, input
   end
+  
+    private
+
+    def self.fetch_obj_space_collection
+      (self.obj_spase_mutex ||= Mutex.new).synchronize do
```

Comment:
ah.. i see that we recreate instance of Pry::InputCompleter every time, perhaps we shouldn't do that. `#call()` takes arguments, but that's probably a bigger change.